packages/node-cli/src/commands/DeployCommandGenerator.js

```diff
@@ -136,60 +137,75 @@ module.exports = class DeployCommandGenerator extends BaseCommandGenerator {
 	}
 
 	async _executeAction(answers) {
-		const SDKParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
-		const flags = [COMMAND.FLAGS.NO_PREVIEW, COMMAND.FLAGS.SKIP_WARNING];
-		if (SDKParams[COMMAND.FLAGS.VALIDATE]) {
-			delete SDKParams[COMMAND.FLAGS.VALIDATE];
-			flags.push(COMMAND.FLAGS.VALIDATE);
+		try {
+			const SDKParams = CommandUtils.extractCommandOptions(answers, this._commandMetadata);
+			const flags = [COMMAND.FLAGS.NO_PREVIEW, COMMAND.FLAGS.SKIP_WARNING];
+			if (SDKParams[COMMAND.FLAGS.VALIDATE]) {
+				delete SDKParams[COMMAND.FLAGS.VALIDATE];
+				flags.push(COMMAND.FLAGS.VALIDATE);
+			}
+			const executionContextForDeploy = new SDKExecutionContext({
+				command: this._commandMetadata.sdkCommand,
+				includeProjectDefaultAuthId: true,
+				params: SDKParams,
+				flags,
+			});
+
+			const operationResult = await executeWithSpinner({
+				action: this._sdkExecutor.execute(executionContextForDeploy),
+				message: TranslationService.getMessage(MESSAGES.DEPLOYING),
+			});
+
+			var isServerValidation = SDKParams[COMMAND.FLAGS.VALIDATE] ? true : false;
+			var isApplyContentProtection =
+				(this._projectType === PROJECT_SUITEAPP && SDKParams[COMMAND.OPTIONS.APPLY_CONTENT_PROTECTION] === SDK_TRUE)
+					? true
+					: false;
+
+			return operationResult.status == ActionResult.SUCCESS
```

Comment:
We should not use ActionResult.SUCCESS in this comparison, because we are checking a OperationResult.status, that actually have the same value, but they are different concepts.
The same happens in the other command generators.

Reply:
ok, I put it in SDKOperationResultUtils for now